packages/haul-core/src/server/setupLiveReload.ts

```diff
@@ -35,11 +36,13 @@ export default function setupLiveReload(
          * React Native client opens connection at `/onchange`
          * and awaits reload signal (http status code - 205)
          */
-        const watcher = request.raw;
+        const watcher = Object.assign(request.raw, { notified: false });
         watchers.push(watcher);
         watcher.req.on('close', () => {
           watchers.splice(watchers.indexOf(watcher), 1);
-          resolve('OK');
+          if (!watcher.notified) {
+            resolve('OK');
+          }
```

Comment:
I think this would also work:
```js
        const index = watchers.indexOf(watcher);
        if (index >= 0) {
          watchers.splice(index, 1);
          resolve('OK');
        }
```

Reply:
I am interested in this project!,
When is this bug fixed on main branch??

Reply:
It will be once I merge the PR and make a release.